src/main/java/com/mraof/minestuck/item/StructureScannerItem.java

```diff
@@ -21,65 +37,112 @@ public class StructureScannerItem extends Item
 {
 	private final TagKey<Structure> structure;
 	private final Supplier<Item> fuelItem;
+	private final String scannerName;
 	
-	public StructureScannerItem(Properties properties, TagKey<Structure> structure, Supplier<Item> fuelItem)
+	public StructureScannerItem(Properties properties, TagKey<Structure> structure, Supplier<Item> fuelItem, String scannerName)
 	{
 		super(properties);
 		this.structure = structure;
 		this.fuelItem = fuelItem;
+		this.scannerName = scannerName;
+		
+		/*
+		ItemProperties.register(this, new ResourceLocation("angle"),
+				new CompassItemPropertyFunction((level, stack, entity) -> stack.hasTag() &&
+						stack.getTag().contains("TargetLocation") ? GlobalPos.of(level.dimension(),
+						NbtUtils.readBlockPos(stack.getTag().getCompound("TargetPos"))) : null));
+		*/
+		
+		ItemProperties.register(this, new ResourceLocation("angle"),
+				new CompassItemPropertyFunction((level, stack, entity) -> {
+					if(stack.hasTag() && stack.getTag().contains("TargetLocation"))
+					{
+						return GlobalPos.of(level.dimension(), NbtUtils.readBlockPos(stack.getTag().getCompound("TargetLocation")));
+					} else
+					{
+						return null;
+					}
+				}));
+		
+		ItemProperties.register(this, new ResourceLocation(Minestuck.MOD_ID, "powered"),
+				((pStack, pLevel, pEntity, pSeed) -> pStack.hasTag() && pStack.getTag().getBoolean("Powered") ? 1 : 0));
```

Comment:
I dont believe these should exist in the constructor. It causes a crash when used with a dedicated server. From memory, it may need to go into ClientProxy instead

Reply:
Correct. Move these to the other `ItemProperties.register()` calls inside `ClientProxy.init()`

Reply:
Ah, I wasn't aware ClientProxy was a Minestuck addition. I was thinking it was a Minecraft component, and thus uneditable. Well that's handy!

Moved the item properties to the Client Proxy init function.